### ISSUE TITLE
v3.0.10 breaks specificity when used with Vue scoped style

### ISSUE THREAD

**Issue by alexcroox** [tailwindlabs/tailwindcss#6895]
**What version of Tailwind CSS are you using?**

For example: v3.0.10

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss-cli 8.4.5, "@nuxt/postcss8": "1.1.3", "nuxt-start": "2.15.8",

**What version of Node.js are you using?**

v14.16.1

**What browser are you using?**

For example: Chrome

**What operating system are you using?**

For example: macOS

**Describe your issue**

After upgrading from v2 to v3 our shadows no longer work locally (so I know it's not a production purge css issue).

I think the utility specific styles are being overridden by global reset ones?

![image](https://user-images.githubusercontent.com/1065835/148196164-a3ffc17c-d005-4546-b49c-fba28cf99e18.png)

```
<div
    class="flex flex-col flex-auto rounded"
    :class="{ 'bg-white shadow': !transparent }"
  >
```

**Comment by alexcroox** [comment]
After more experimentation it seems this occurs whenever I have `<style lang="scss" scoped>` in the component or component's parent. Very strange. If I remove the `scoped` property it resolves itself.

**Comment by christhofer** [comment]
This also happens to one of my company project.
But I still can't replicate this issue with fresh vite + vue + tailwindcss install.

I also suspect the <style scoped> somehow triggers the reset
And because [data-v-somecode] has higher specificity, it override tailwind .shadow class.

**Comment by alexcroox** [comment]
@christhofer have you tried downgrading to v3.0.9?

**Comment by daan33** [comment]
Same here, even without specifying `lang="scss"`.  

I was wondering why some of my borders changed their color after updating TailwindCSS to `3.0.10` this morning. Downgrading to `3.0.9` fixes it.

**Comment by ednar28** [comment]
btw it's working for me when I'm using tailwindcss v.3.0.9

**Comment by christhofer** [comment]
Done with the repro

https://github.com/christhofer/tw-repro

Run `npm run dev`
And then go to src/components/LogoVue.vue
Uncomment the style, save, the shadow somehow disappear

**Comment by luisorbaiceta** [comment]
This is happening too with Next.js, seems that some classes break the build in css scoped modules. In my cases removing the `shadow-md` (or any kind of shadow) fixed the issue in a minimal repro

Rolling back to `v3.0.9` fixes the problem

In case it helps to narrow down the issue

**Comment by reinink** [comment]
Hey all, thanks for reporting this 🙏

We're working on a proper fix for this, but in meantime we've rolled back the change the caused this issue and have released a `v3.0.11` version.
